hera_sim/tests/test_antpos.py

```diff
@@ -1,94 +1,97 @@
+            ants = antpos.hex_array(hex_num, split_core=False, outriggers=Nrings)
+            assert check_ant_labels(Nants_expected=Nants, ants=ants)
+
+
+def test_hex_array_split_core_nants_scaling():
+    for hex_num in range(2, 15):
+        Nants_core = 3 * hex_num * (hex_num - 1) + 1
+        Nants = Nants_core - hex_num
+        ants = antpos.hex_array(hex_num, split_core=True, outriggers=0)
+        assert check_ant_labels(Nants_expected=Nants, ants=ants)
+
+
+def test_hex_array_7_element_positions():
+    for sep in np.linspace(1, 10, 10):
+        ants = antpos.hex_array(2, split_core=False, outriggers=0, sep=sep)
+        # Central antenna is number 3; all should be equidistant from 3.
         for i in range(7):
             if i == 3:
                 continue
-            bl = xyz[i] - xyz[3]
-            bl_len = np.sqrt(np.dot(bl, bl))
-            self.assertAlmostEqual(bl_len, 2, 10)
-        # Test top corner of a lot of configs
-        for N in range(2, 15):
-            xyz = antpos.hex_array(N, split_core=False, outriggers=0, sep=1.0)
-            bl1 = xyz[1] - xyz[0]
-            bl2 = xyz[N] - xyz[0]
-            np.testing.assert_allclose(bl1, [1, 0, 0], atol=1e-10)
-            np.testing.assert_allclose(bl2, [-0.5, -(3 ** 0.5) / 2, 0], atol=1e-10)
-        # Test split (this is rather brittle and could be improved)
-        xyz = antpos.hex_array(3, sep=14.6, split_core=True, outriggers=0)
-        np.testing.assert_allclose(
-            xyz[3] - xyz[7], np.array([0, 2 * 14.6 / 3 ** 0.5, 0])
-        )
-        np.testing.assert_allclose(
-            xyz[5] - xyz[10], np.array([-14.6, 14.6 / 3 ** 0.5, 0])
-        )
-        # Test outriggers (this is rather brittle and could be improved)
-        xyz = antpos.hex_array(11, sep=14.6, split_core=True, outriggers=2)
-        np.testing.assert_allclose(xyz[348] - xyz[347], np.array([10 * 14.6, 0, 0]))
-        np.testing.assert_allclose(
-            xyz[347] - xyz[346], np.array([10 * 14.6, 14.6 / 3 ** 0.5, 0])
-        )
-        np.testing.assert_allclose(
-            xyz[325] - xyz[324], np.array([10 * 14.6, -14.6 / 3 ** 0.5, 0])
-        )
-
-
-if __name__ == "__main__":
-    unittest.main()
+            bl = ants[i] - ants[3]
+            bl_len = np.linalg.norm(bl)
+            assert np.isclose(bl_len, sep)
+
+
+def test_hex_array_core_corner_positions():
+    for sep in np.linspace(1, 10, 10):
+        for hex_num in range(2, 15):
+            ants = antpos.hex_array(hex_num, split_core=False, outriggers=0, sep=sep)
+            bl1 = ants[1] - ants[0]
+            bl2 = ants[hex_num] - ants[0]
+            expected_bl1 = sep * np.array([1, 0, 0])
+            expected_bl2 = sep * np.array([-0.5, -np.sqrt(3) / 2, 0])
+            assert np.allclose(bl1, expected_bl1)
+            assert np.allclose(bl2, expected_bl2)
+
+
+def test_hex_array_split_core_positions():
+    # This test is rather brittle and could be improved.
```

Comment:
the only thing I can see to do is to set `sep` as a local variable

Reply:
I'm not sure if I quite understand what you mean. Would that entail something like perhaps parametrizing `sep` and checking that it works for different base separations? I think what I was getting at with the comment is that it seems like this test assumes a certain ordering of the antennas and basically checks that the ordering is obeyed and that the antennas are placed on a hexagonal grid (and I'd hope that it also tests that there are deviations from a perfect hexagonal grid by looking at baselines that use antennas in different sectors, but tbh I haven't dug quite that deep into validating these tests). I think this is fine as long as the algorithm for building the hex array doesn't change, but it kind of bothers me that it's so specific. Maybe it's not worth worrying over, though. What do you think?

Reply:
I think tests that are specific are fine -- if the code ever changes, the test will break. Then we'll be reminded that we broke API, and have to update documentation and version accordingly.

Reply:
Ok, I can get on board with that. Do you think there are any changes I should make to this and the HERA-350 test?